nipst/nipst_builder.go

```diff
@@ -0,0 +1,221 @@
+package nipst
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"sync"
+	"time"
+)
+
+type proof struct{}
+
+type postCommitment struct {
+	Root  common.Hash
+	Proof proof
+}
+
+func (p *postCommitment) Valid() bool {
+	return true
+}
+
+//todo: figure out which fields are needed
+type MembershipProof struct {
+	Root  common.Hash
+	Proof proof
+}
+
+func (p *MembershipProof) Valid() bool {
+	return true
+}
+
+//todo: poet proof
+type PoetProof struct {
+	Root  common.Hash
+	Proof proof
+}
+
+func (p *PoetProof) Valid() bool {
+	return true
+}
+
+type PoetService interface {
+	SendPostProof(root common.Hash, p proof) error
+	GetMembershipProof(root common.Hash, timeout time.Duration) (*MembershipProof, error)
+	GetPoetProof(root common.Hash, timeout time.Duration) (*PoetProof, error)
+}
+
+type PostService interface {
+	SendChallenge(root common.Hash, challenge common.Hash) error
+	ChallengeOutput(root common.Hash, challenge common.Hash, timeout time.Duration) (*postCommitment, error)
+}
+
+type ActivationBuilder interface {
+	BuildActivationTx(proof Nipst)
+}
+
+type Nipst struct {
+	initialPost     *postCommitment
+	membershipProof *MembershipProof
+	poetProof       *PoetProof
+	secondPost      *postCommitment
+}
+
+type NipstBuilder struct {
```

Comment:
This is a builder of many NIPST instances, so the name is a bit confusing. Perhaps `nipstsBuilder`?

Reply:
This means changing BlockListener, TxListener and BlockBuilder to plurals, I'd rather not do this ruight now